obspy/signal/cross_correlation.py

```diff
@@ -26,14 +26,166 @@
+
+    :return: *shift, value, corr_fun* for ``full_xcorr=False`` and
+             only *shift, value* for ``full_xcorr=False`` (depreciated).
+             *shift* is the sample index of the maximum correlation
+             relative to zero shift (the middle). If this is nonunique the
+             smallest value of *shift* will be returned.
+             *value* is the value at this sample.
+             *corr_fun* is the complete cross-correlation function.
+
+    .. note::
+
+        For most input parameters cross-correlation in frequency domain is much
+        faster.
+        Only for small values of ``shift_len`` time domain cross-correlation
+        migth save some time (``shifth_len⪅100``).
+
+    .. note::
+
+        If the signals have different length, they will be aligned around
+        their middle. The sample with zero shift in the cross-correlation
+        function corresponds to this correlation:
+
+        ::
+
+            --aaaa--
+            bbbbbbbb
+
+        For odd ``len(tr1)-len(tr2)`` the cross-correlation function will
+        consist of only ``2*shift_len`` samples because a shift of 0
+        corresponds to the middle between two samples. *shift* will be a float
+        value in this case (otherwise int).
+
+    .. rubric:: Example
+
+    >>> tr1 = np.random.randn(10000).astype(np.float32)
+    >>> a, b, x = xcorr(tr1, tr1, 1000, full_xcorr=True)
+    >>> a
+    0
+    >>> round(b, 6)
+    1.0
+    >>> tr2 = np.roll(tr1, 50)  # shift tr2 by 50 samples
+    >>> a, b, x = xcorr(tr1, tr2, 1000, full_xcorr=True)
+    >>> a
+    -50
+    >>> round(b, 6)
+    0.995369
+
+    """
+    a, b = tr1, tr2
+    # if we get Trace objects, use their data arrays
+    if isinstance(a, Trace):
+        a = a.data
+    if isinstance(b, Trace):
+        b = b.data
+    a = np.asarray(a)
+    b = np.asarray(b)
+    if demean:
+        a = a - np.mean(a)
+        b = b - np.mean(b)
+    if normalize:
+        stdev = (np.sum(a ** 2)) ** 0.5 * (np.sum(b ** 2)) ** 0.5
+        if stdev == 0:
+            # set stdev to 1 to prevent division by 0
+            # cross-correlation function will have only zeros anyway
+            stdev = 1
+    else:
+        stdev = 1
+    # choose the usually faster xcorr method for each domain
+    _xcorr = _xcorr_slice if domain == 'freq' else _xcorr_padzeros
+    c = _xcorr(a, b, shift_len, domain=domain) / stdev
+    shift, value = xcorr_max(c, abs_max=abs_max)
+    if not full_xcorr:
+        from obspy.core.util.deprecation_helpers import ObsPyDeprecationWarning
+        msg = ('Keyword full_xcorr will default to True starting with the next'
+               ' major release (v1.2) and will be removed in the subsquent '
+               'major release (v.1.3). Please set full_xcorr=True.')
+        warnings.warn(msg, ObsPyDeprecationWarning)
+        return shift, value
+
+    return shift, value, c
+
+
+def _xcorr_old_implementation(tr1, tr2, shift_len, full_xcorr=False):
```

Comment:
Maybe put a deprecation warning as first line inside this function? I don't think we want to drag around tons of redundant implementations forever..?